events/include/gz/common/Event.hh

```diff
@@ -93,39 +151,37 @@ namespace gz
 
       /// \brief Friend class.
       public: template<typename T, typename N> friend class EventT;
+      public: friend class Event;
     };
 
     /// \brief A class for event processing.
     /// \tparam T function event callback function signature
     /// \tparam N optional additional type to disambiguate events with same
     ///   function signature
+    //
+    // WARNING: This class should not have any data members. Instead all data
+    // members should go into the base class `Event`. This is to ensure that
+    // this class behaves as expected when used with shared libraries (plugins)
+    // that get unloaded after emitting an event.
+    //
+    // Explanation: TODO
+    //
     template<typename T, typename N = void>
-    class EventT : public Event
+    class EventT final : public Event
```

Comment:
Are we sure that no one is currently inheriting from `EventT<T>`?  I can't recall anywhere in our libraries that we are doing it.

Reply:
I don't know. As in my previous comment, I made it `final` to prevent users from inheriting from this and adding data members. We can drop the `final` and just warn users not to add data members if they inherit from `EventT`, but that doesn't seem reliable.